index.js

```diff
@@ -687,12 +705,12 @@ module.exports = function (log, indexesPath) {
     offset = offset || 0
     const start = Date.now()
 
-    const timestamped = bitset.array().map((o) => {
-      return {
-        offset: o,
-        timestamp: indexes['timestamp'].tarr[o],
-      }
-    })
+    const timestampTarr = indexes['timestamp'].tarr
+    const timestamped = bitset.array()
+    for (let i = 0, len = timestamped.length; i < len; ++i) {
+      const o = timestamped[i]
+      timestamped[i] = { offset: o, timestamp: timestampTarr[o] }
+    }
     const sorted = timestamped.sort((a, b) => {
```

Comment:
I'm not sure about this one, its a bit hard to read because timestamp is something else first (the .array line) and then becomes something completely different.

Reply:
Yeah I can revert this one back, I agree it was an aggressive one